docs/deprecated-sql-function.md

```diff
@@ -0,0 +1,195 @@
+                array_remove(primary_source_columns, 'the_geom'),
+                -- And we add our new `the_geom` column using the `radius` argument.
+                ARRAY['ST_Buffer(the_geom::geography, ' || radius || ')::geometry AS the_geom']
+              ),
+              ','
+            ) INTO selected_columns;
+            IF operation = 'create' THEN
+                -- In some cases you might want to drop the table.
+                -- EXECUTE 'DROP TABLE ' || table_name || ';';
+                -- We use our prepared columns to select from the original query.
+                EXECUTE 'CREATE TABLE ' || table_name || ' AS SELECT ' || selected_columns || ' FROM (' || primary_source_query || ') _q LIMIT 0';
+            ELSIF operation = 'populate' THEN
+                EXECUTE 'INSERT INTO ' || table_name || ' SELECT ' || selected_columns || ' FROM (' || primary_source_query || ') _q';
+            END IF;
+        END;
+$$ LANGUAGE plpgsql;
+```
+
+The framework, behind the scenes, will call that function in two phases:
+
+1. It call the new function with the `create` operation, so we can create a table, something like:
+
+```sql
+select * from DEP_EXT_buffer(
+    'create',
+    'wadus_table',
+    'select 1 as cartodb_id, st_setsrid(st_makepoint(0,0), 4326) as the_geom',
+    ARRAY['cartodb_id', 'the_geom'],
+    1000
+);
+```
+
+2. Later, it will invoke the same function but it will do it with `populate` as the operation to perform, like in:
+
+```sql
+select * from DEP_EXT_buffer(
+    'populate',
+    'wadus_table',
+    'select 1 as cartodb_id, st_setsrid(st_makepoint(0,0), 4326) as the_geom',
+    ARRAY['cartodb_id', 'the_geom'],
+    1000
+);
+```
+
+### Spatial Interpolation as in https://github.com/CartoDB/camshaft/pull/153
+
+```sql
+CREATE OR REPLACE FUNCTION DEP_EXT_SpatialInterpolation(
+        operation text,
+        table_name text,
+        source_query text, source_columns text[],
+        target_query text, target_columns text[],
+        val_column text,
+        method numeric, -- 0=nearest neighbor, 1=barymetric, 2=IDW
+        number_of_neighbors numeric, -- DEFAULT 0, -- 0=unlimited
+        decay_order numeric -- DEFAULT 0,
+    )
+    RETURNS VOID AS $$
+        BEGIN
+            IF method < 0 OR method > 2 THEN
+                RAISE EXCEPTION 'Invalid method value=%', buster
+                    USING HINT = 'Valid ones are: 0=nearest neighbor, 1=barymetric, 2=IDW';
+            END IF;
+            IF operation = 'create' THEN
+                EXECUTE 'CREATE TABLE ' || table_name || ' AS ' ||
+                    -- The table will have the schema of target query but with an extra column from the source query.
+                    'SELECT _target.*, _source.' || val_column ||
+                    ' FROM (' || target_query || ') _target, (' || source_query || ') _source LIMIT 0';
+            ELSIF operation = 'populate' THEN
+                EXECUTE 'INSERT INTO ' || table_name || ' ' ||
```

Comment:
After reading the documentation, there is a limitation following that approach:

> Note that parameter symbols can only be used for data values — if you want to use dynamically determined table or column names, you must insert them into the command string textually.

I'm gonna try to rework the example to improve it, but for those scenarios, we need to build the string manually.